### PR TITLE
add update time to FieldIndex

### PR DESCRIPTION
**Pull request by thebrianchen**
Incremental PR to add storing and retrieving last update time from FieldIndex. Also renamed collectionId to collectionGroup on fieldIndex.

### CODE REVIEW

File: firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java [hunk 1]
```diff
@@ -265,6 +267,19 @@ public void testCollectionGroup() {
...
+    indexManager.addFieldIndex(
+        new FieldIndex("coll1")
+            .withAddedField(field("value"), FieldIndex.Segment.Kind.ORDERED)
+            .withUpdateTime(new Timestamp(10, 20)));
+
+    List<FieldIndex> indexes = ((SQLiteIndexManager) indexManager).getFieldIndexes();
+    assertEquals(indexes.size(), 1);
+    FieldIndex index = indexes.get(0);
+    assertEquals(index.getUpdateTime(), new Timestamp(10, 20));
+  }
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java:281]: This test seems better suited for our serialization tests.

**Review by schmidt-sebastian — COMMENTED** [review]
Some feedback. Please send to @wu-hui if you have updates by Monday.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java [hunk 2]
```diff
@@ -311,8 +311,9 @@ public Index encodeFieldIndex(FieldIndex fieldIndex) {
     return index.build();
   }
 
-  public FieldIndex decodeFieldIndex(String collection_group, int indexId, Index index) {
-    FieldIndex fieldIndex = new FieldIndex(collection_group, indexId);
+  public FieldIndex decodeFieldIndex(
+      String collectionGroup, int indexId, Index index, int updateSeconds, int updateNanos) {
```
[the remark this answers is not in the document]
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java:315]: done.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldIndex.java [hunk 3]
```diff
@@ -118,18 +131,19 @@ public boolean equals(Object o) {
     FieldIndex fieldIndex = (FieldIndex) o;
 
     if (!segments.equals(fieldIndex.segments)) return false;
-    return collectionId.equals(fieldIndex.collectionId);
+    return collectionGroup.equals(fieldIndex.collectionGroup);
```
[the remark this answers is not in the document]
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldIndex.java:134]: done.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldIndex.java [hunk 4]
```diff
@@ -118,18 +131,19 @@ public boolean equals(Object o) {
...
   }
 
   @Override
   public int hashCode() {
-    int result = collectionId.hashCode();
+    int result = collectionGroup.hashCode();
```
[the remark this answers is not in the document]
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldIndex.java:139]: done.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldIndex.java [hunk 5]
```diff
@@ -118,18 +131,19 @@ public boolean equals(Object o) {
...
     result = 31 * result + segments.hashCode();
     return result;
   }
 
   @Override
   public String toString() {
-    return String.format("FieldIndex{collectionId='%s', segments=%s}", collectionId, segments);
+    return String.format(
+        "FieldIndex{collectionGroup='%s', segments=%s}", collectionGroup, segments);
```
[the remark this answers is not in the document]
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/model/FieldIndex.java:147]: done.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java (hunk 1 above)
Comment (thebrianchen) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java:281]: The test here checks that the persistence queries in `addFieldIndex` and `getFieldIndex` are correct.

**Review by thebrianchen — COMMENTED** [review]
thanks for the speedy review!

**Review by wu-hui — COMMENTED** [review]
@thebrianchen You replied some comments, but the actual changes are not applied it seems. Did you forget to push a new commit?

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java [hunk 6]
```diff
@@ -370,4 +383,28 @@ private boolean isMultiValueFilter(Target target, FieldPath fieldPath) {
   private byte[] encodeFieldIndex(FieldIndex fieldIndex) {
     return serializer.encodeFieldIndex(fieldIndex).toByteArray();
   }
+
+  // TODO(indexing): Add support for fetching last N updated entries.
+  public List<FieldIndex> getFieldIndexes() {
+    List<FieldIndex> allIndexes = new ArrayList<>();
+    db.query(
+            "SELECT index_id, collection_group, index_proto, update_time_seconds, update_time_nanos FROM index_configuration "
+                + "WHERE active")
```
Comment (wu-hui) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java:392]: Does `WHERE active` actually work? And even if it works, we should still be using `active = 1`.

**Review by wu-hui — CHANGES_REQUESTED** [review]

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java (hunk 6 above)
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java]: Thanks for catching.  SQL query still works without adding `= 1`, but it's best to follow best practices.

**Review by wu-hui — APPROVED** [review]

**Pull request merged**